### PR TITLE
Read in grid functions to objects

### PR DESCRIPTION
**Pull request by jonnyh404**

### CODE REVIEW

File: ResSimpy/Nexus/array_function_operations.py [hunk 1]
```diff
@@ -289,3 +292,114 @@ def summarize_model_functions(function_list_to_parse: list[list[str]]) -> pd.Dat
     function_summary_df = function_summary_df.set_index('FUNCTION #')
 
     return function_summary_df
+
+
+def create_grid_array_function_objects(array_functions_as_list: list[list[str]]) -> list[NexusGridArrayFunction]:
+    store_array_functions: list[NexusGridArrayFunction] = []
+    for function_number, array_function in enumerate(array_functions_as_list, 1):
```
Comment (kwr3) [inline ResSimpy/Nexus/array_function_operations.py:299]: consider comment describing what the second argument of enumerate() does

File: ResSimpy/Nexus/array_function_operations.py [hunk 2]
```diff
@@ -289,3 +292,114 @@ def summarize_model_functions(function_list_to_parse: list[list[str]]) -> pd.Dat
...
+        new_grid_array_function_obj = object_from_array_function_block(array_function, function_number)
+        store_array_functions.append(new_grid_array_function_obj)
+    return store_array_functions
+
+
+def object_from_array_function_block(array_function: list[str], function_number: int) -> NexusGridArrayFunction:
```
Comment (kwr3) [inline ResSimpy/Nexus/array_function_operations.py:305]: A rather beefy method. A doc string would be good here or might consider breaking it up into chunks.

File: ResSimpy/Nexus/array_function_operations.py [hunk 3]
```diff
@@ -289,3 +292,114 @@ def summarize_model_functions(function_list_to_parse: list[list[str]]) -> pd.Dat
...
+    input_range_list = None
+    output_range_list = None
+    drange_list = None
+    function_type_enum = None
+
+    for li, line in enumerate(array_function):
+        modified_line = line.upper()
+        words = modified_line.split()
+        if 'FUNCTION' in modified_line:
+            if len(words) == 1:
```
Comment (kwr3) [inline ResSimpy/Nexus/array_function_operations.py:323]: Comments are good to include with the use of continue and break statements. Why do we skip iterations with words with a length of 1

**Review by kwr3 — APPROVED** [review]
Minor suggestions to consider.

File: ResSimpy/Nexus/array_function_operations.py [hunk 4]
```diff
@@ -289,3 +312,173 @@ def summarize_model_functions(function_list_to_parse: list[list[str]]) -> pd.Dat
     function_summary_df = function_summary_df.set_index('FUNCTION #')
 
     return function_summary_df
+
+
+def create_grid_array_function_objects(array_functions_as_list: list[list[str]]) -> list[NexusGridArrayFunction]:
```
Comment (cflynn3) [inline ResSimpy/Nexus/array_function_operations.py:317]: Please add a docstring for this public method.

**Review by cflynn3 — APPROVED** [review]
Looks good, just one minor comment.

**Pull request merged**